cloud/scope/machinepoolmachine.go

```diff
@@ -318,15 +322,24 @@ func (m *MachinePoolMachineScope) HasLatestModelApplied(_ context.Context, insta
 	// Get the image from the disk URL path to compare with the latest image name
 	diskImage, err := url.Parse(instance.SourceImage)
 	if err != nil {
+		log.Error(err, "Error Parsing the Instance.SourceImage from the disk attached.")
 		return false, err
 	}
 	instanceImage := path.Base(diskImage.Path)
 
 	// Check if the image is the latest
-	if image == instanceImage {
+	// Check if the Labels applied to the instance are the latest as in the Instance Template.
+	// Remove the two keys of `capg-role` and `capg-cluster-<CLUSTER-NAME>` as they are added by default by CAPG
+	// and would be easier to compare the additional labels.
+	// ref: https://github.com/newrelic-forks/cluster-api-provider-gcp/blob/ef2e7f1e64ebeeb5389c446fe4cf89026fcb8a8a/cloud/services/compute/instances/reconcile_test.go#L244-L245
+	delete(instanceLabelsToCompare, "capg-role")
```

Comment:
I wonder if removing the labels adds any value assuming if they are always present by default.

Reply:
resolving this comment, as this logic has been refactored.